### PR TITLE
remove dependency on openshift struct

### PR DESCRIPTION
**Pull request by baude**
use of github.com/openshift/api/config/v1 is costly in size, something around 10MB.  removal of this dependancy and using a locally defined struct took the buildah executable size from around 38MB to around 27MB.  This also hits libpod and allows it also shed that size as well.

Signed-off-by: Brent Baude <bbaude@redhat.com>

### CODE REVIEW

File: commit.go [hunk 1]
```diff
@@ -181,7 +180,11 @@ func checkRegistrySourcesAllows(forWhat string, dest types.ImageReference) error
 	}
 
 	if registrySources, ok := os.LookupEnv("BUILD_REGISTRY_SOURCES"); ok && len(registrySources) > 0 {
-		var sources configv1.RegistrySources
+		var sources struct {
```
Comment (mheon) [inline commit.go:183]: Should at least mention where this came from

**Comment by TomSweeneyRedHat** [comment]
Changes LGTM, tests aren't happy at the moment, but I don't think it's related.  Looks like Friday afternoon CI going on vacation.

My only small concern is these definitions diverging at some point, but I think there's an extremely low chance of that happening.

**Comment by TomSweeneyRedHat** [comment]
bors retest

**Comment by rhatdan** [comment]
Incredible.  Great work @baude 
bors r+
@TomSweeneyRedHat Lets cut a minor release as soon as this passes.

**Comment by TomSweeneyRedHat** [comment]
@rhatdan if possible, I'd like to get my imagebuilder fix in too.

**Comment by vrothberg** [comment]
Sweet! Nice work, @baude :+1:

**Pull request merged**